tests/Ticket/1783TestCase.php

```diff
@@ -12,9 +12,12 @@ public function testValidateLargeIntegers()
         $this->manager->setAttribute(Doctrine_Core::ATTR_VALIDATE, Doctrine_Core::VALIDATE_ALL);        
 
         $test = new Ticket_1783();
+
         $test->bigint = PHP_INT_MAX + 1;
-        
-        $this->assertTrue($test->isValid());
+
+        // This test works on php 32bit version because float allow to represent bigger value than a int.
+        // On 64bit, int is now equivalent to a database storage of a bigint
+        $this->assertTrue((PHP_INT_MAX == 2147483647) ? $test->isValid() : true);
```

Comment:
Yes, nice catch, it is a good track, but the conclusion is wrong.

The table definition will correspond to MySql type `BIGINT UNSIGNED`.

Which has a maximum value of `2^64 - 1`.
However, `PHP_INT_MAX` on 64bit system architecture is `2^63-1` as PHP integer is not unsigned.

Using `getErrorStackAsString()` we can see what validator failed.
We can see that the error comes from `unsigned` validation.

* More over, there is no validation on the length for integer.
    https://github.com/FriendsOfSymfony1/doctrine1/blob/master/lib/Doctrine/Validator.php#L95-L96
* Digging deeper of the `Unsigned` validator, you can see that the regex is in fault.
    It depends on how the value is converted to string, when it is bigger than PHP_INT_MAX it contains `E` the exponential symbol.

**Conclusion**
The issue still exists for 64bit system.

For now, we can ignore this test.
https://github.com/FriendsOfSymfony1/doctrine1/pull/56/commits/6a08c161467bddf89afd0eeeacfa6abf085d2c47